File: finstack/valuations/tests/golden/runners/integration_rates.rs
//! Domain runner for rates calibrate-then-price integration goldens.

use crate::golden::runner::DomainRunner;
use crate::golden::schema::GoldenFixture;
use std::collections::BTreeMap;

/// Rates integration golden runner.
pub struct IntegrationRatesRunner;

impl DomainRunner for IntegrationRatesRunner {
    fn run(&self, fixture: &GoldenFixture) -> Result<BTreeMap<String, f64>, String> {
        crate::golden::runners::reject_flattened_outputs("rates integration runner", fixture)
    }
}

File: finstack-py/tests/golden/runners/integration_rates.py
"""Domain runner for rates integration golden fixtures."""

from __future__ import annotations

from tests.golden.runners._placeholders import reject_flattened_outputs
from tests.golden.schema import GoldenFixture


def run(fixture: GoldenFixture) -> dict[str, float]:
    """Require executable rates integration inputs before running."""
    return reject_flattened_outputs("rates integration runner", fixture)
